pkg/chunk/local/fs_object_client.go

```diff
@@ -95,53 +95,60 @@ func (f *FSObjectClient) PutObject(ctx context.Context, objectKey string, object
 	return fl.Close()
 }
 
-// List objects and common-prefixes i.e directories from the store non-recursively
-func (f *FSObjectClient) List(ctx context.Context, prefix string) ([]chunk.StorageObject, []chunk.StorageCommonPrefix, error) {
+// FSObjectClient assumes that prefix is a directory, and only supports "" and "/" delimiters.
+func (f *FSObjectClient) List(ctx context.Context, prefix, delimiter string) ([]chunk.StorageObject, []chunk.StorageCommonPrefix, error) {
+	if delimiter != "" && delimiter != "/" {
+		return nil, nil, fmt.Errorf("unsupported delimiter: %q", delimiter)
+	}
+
 	var storageObjects []chunk.StorageObject
 	var commonPrefixes []chunk.StorageCommonPrefix
 
-	folderPath := filepath.Join(f.cfg.Directory, prefix)
+	folderPath := filepath.Join(f.cfg.Directory, filepath.FromSlash(prefix))
+	err := filepath.Walk(folderPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
 
-	_, err := os.Stat(folderPath)
-	if err != nil {
-		if os.IsNotExist(err) {
```

Comment:
I think we should preserve this check to be consistent with other stores where it is not an error to call list with a prefix which does not exist.
What do you think?

Reply:
Good point! Fixed now, PTAL.